File: ios/Packages/FeatureOrders/Sources/FeatureOrders/Data/OrdersRepositoryImpl.swift
import Foundation

/// Stub repository — returns hardcoded data for scaffolding.
public final class OrdersRepositoryImpl: OrdersRepository {
    public init() {}

    public func getOrders() async throws -> [Order] {
        [
            Order(id: "1", title: "Order #1001", amount: 45.00, status: .completed),
            Order(id: "2", title: "Order #1002", amount: 120.50, status: .processing),
            Order(id: "3", title: "Order #1003", amount: 89.99, status: .pending),
        ]
    }
}

File: flutter/feature_orders/lib/src/data/orders_repository_impl.dart
import '../domain/order.dart';
import '../domain/orders_repository.dart';

/// Stub repository — returns hardcoded data for scaffolding.
class OrdersRepositoryImpl implements OrdersRepository {
  @override
  Future<List<Order>> getOrders() async {
    return const [
      Order(id: '1', title: 'Order #1001', amount: 45.00, status: OrderStatus.completed),
      Order(id: '2', title: 'Order #1002', amount: 120.50, status: OrderStatus.processing),
      Order(id: '3', title: 'Order #1003', amount: 89.99, status: OrderStatus.pending),
    ];
  }
}
